src/ProjectSystem/Impl/Extensions/ProjectTreeExtensions.cs

```diff
@@ -63,6 +63,38 @@ public static class ProjectTreeExtensions {
             return Enumerable.Empty<string>();
         }
 
+        public static IEnumerable<string> GetAllFilePaths(this IEnumerable<IProjectTree> nodes) {
+            if (nodes != null) {
+                List<string> paths = new List<string>();
+                foreach (IProjectTree node in nodes) {
+                    if (node.IsFolder) {
+                        paths.AddRange(node.GetAllFilePaths());
+                    } else if (node.Children.Count > 0) {
+                        paths.AddRange(node.Children.GetAllFilePaths());
+                    } else if (!string.IsNullOrWhiteSpace(node.FilePath)) {
+                        paths.Add(node.FilePath);
+                    }
+                }
+                return paths;
+            }
+            return Enumerable.Empty<string>();
+        }
+
+        public static IEnumerable<string> GetAllFilePaths(this IProjectTree node) {
```

Comment:
No this case is to handle folder nodes. You can do multi select where you select a bunch of R files and then select a bunch of folders. This code picks up all the files under the folder nodes. Without this, you only get the files you selected.

---
In reply to: [80395305](https://github.com/Microsoft/RTVS/pull/2405#discussion_r80395305) [](ancestors = 80395305)